components/script/dom/activation.rs

```diff
@@ -48,43 +48,32 @@ pub trait Activatable {
     }
 }
 
-/// Whether an activation was initiated via the click() method
-#[derive(PartialEq)]
-pub enum ActivationSource {
-    FromClick,
-    NotFromClick,
-}
-
-// https://html.spec.whatwg.org/multipage/#run-synthetic-click-activation-steps
+//https://html.spec.whatwg.org/multipage/#fire-a-synthetic-mouse-event
 pub fn synthetic_click_activation(
     element: &Element,
     ctrl_key: bool,
     shift_key: bool,
     alt_key: bool,
     meta_key: bool,
-    source: ActivationSource,
+    not_trusted: bool,
 ) {
-    // Step 1
-    if element.click_in_progress() {
-        return;
-    }
-    // Step 2
-    element.set_click_in_progress(true);
-    // Step 3
+    // Step 12 of https://dom.spec.whatwg.org/#concept-event-dispatch
     let activatable = element.as_maybe_activatable();
     if let Some(a) = activatable {
         a.pre_click_activation();
```

Comment:
As mentioned in [@jdm's comment](https://github.com/servo/servo/issues/22783#issuecomment-469793779), this stuff needs to be moved into the event dispatch code

https://dom.spec.whatwg.org/#concept-event-dispatch

Reply:
I'll remove this from here.
Coming to implementation in the event dispatch code there's a [comment](https://github.com/servo/servo/blob/d0ac1742bbe9425e8418c0884d461fd786aba35c/components/script/dom/event.rs#L187) mentioning step 12 of the [spec](https://dom.spec.whatwg.org/#concept-event-dispatch) . Although I am unsure of the correctness of this [implementation](https://github.com/servo/servo/blob/d0ac1742bbe9425e8418c0884d461fd786aba35c/components/script/dom/event.rs#L208).